### ISSUE TITLE
'useImportType' rule incompatibilty with Angular constructors dependecy injection

### ISSUE THREAD

**Issue by vnabet** [biomejs/biome#2309]
### Environment information

```bash
CLI:
  Version:                      1.6.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v18.18.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/9.8.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:
Workspace:
  Open Documents:               0
```

### Rule name

useImportType

### Playground link

https://github.com/vnabet/ng-test-biome

### Expected result

'useImportType' rule incompatibilty with Angular constructors dependecy injection

This code is ok for recommended biome rules, but it can't work
![Capture d'écran 2024-04-04 170223](https://github.com/biomejs/biome/assets/1830490/c21e2c03-2d9f-4869-a8a0-9afbce662cf2)

Linting error but this code is correct
![Capture d'écran 2024-04-04 171026](https://github.com/biomejs/biome/assets/1830490/f3516f5f-fe70-4580-9a63-ebdbcfbfb1be)

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Same issue as #2003 and #2261.

We decided to follow the [decision of TypeScript ESLint](https://github.com/biomejs/biome/issues/2003#issuecomment-2028665621).

**Comment by ematipico** [comment]
@Conaclos can we document "this decision" in the rule, please?
